xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -1005,14 +1031,80 @@ private static FilterChainMatch parseFilterChainMatch(
         }
         // TODO(chengyuanzhang): validate if the sum of weights equals to total weight.
         return StructOrError.fromStruct(RouteAction.forWeightedClusters(
-            weightedClusters, hashPolicies, timeoutNano));
+            weightedClusters, hashPolicies, timeoutNano, retryPolicy));
       case CLUSTERSPECIFIER_NOT_SET:
       default:
         return StructOrError.fromError(
             "Unknown cluster specifier: " + proto.getClusterSpecifierCase());
     }
   }
 
+  @Nullable // Return null if we ignore the given policy.
+  private static StructOrError<RetryPolicy> parseRetryPolicy(
+      io.envoyproxy.envoy.config.route.v3.RetryPolicy retryPolicyProto) {
+    int maxAttempts = 2;
+    if (retryPolicyProto.hasNumRetries()) {
+      maxAttempts = retryPolicyProto.getNumRetries().getValue() + 1;
+    }
+    Duration initialBackoff = Durations.fromMillis(25);
+    Duration maxBackoff = Durations.fromMillis(250);
+    if (retryPolicyProto.hasRetryBackOff()) {
+      RetryBackOff retryBackOff = retryPolicyProto.getRetryBackOff();
+      if (!retryBackOff.hasBaseInterval()) {
+        return StructOrError.fromError("No base_interval specified in retry_backoff");
+      }
+      Duration originalInitialBackoff = initialBackoff = retryBackOff.getBaseInterval();
+      if (Durations.compare(initialBackoff, Durations.ZERO) <= 0) {
+        return StructOrError.fromError("base_interval in retry_backoff must be positive");
+      }
+      if (Durations.compare(initialBackoff, Durations.fromMillis(1)) < 0) {
+        initialBackoff = Durations.fromMillis(1);
+      }
+      if (retryBackOff.hasMaxInterval()) {
+        maxBackoff = retryPolicyProto.getRetryBackOff().getMaxInterval();
+        if (Durations.compare(maxBackoff, originalInitialBackoff) < 0) {
```

Comment:
Just to confirm: the initial backoff must be strictly less. It isn't permitted to be equal.

Reply:
They could be equal:

https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-retrypolicy-retrybackoff

```
max_interval

(Duration) Specifies the maximum interval between retries. 
This parameter is optional, but must be greater than or equal to the base_interval if set. 
```

Reply:
And I don't see any restriction on our original gRFC.

Reply:
Oh, nm. I misread it; I didn't invert it in my mind. I thought we weren't permitting equal.